arduino/builder/sketch.go

```diff
@@ -58,6 +59,44 @@ func SketchSaveItemCpp(item *sketch.Item, destPath string) error {
 	return nil
 }
 
+func simpleLocalWalkRecursive(root string, maxDepth int, walkFn func(path string, info os.FileInfo, err error) error) error {
+
+	info, err := os.Stat(root)
+
+	if err != nil {
+		return walkFn(root, nil, err)
+	}
+
+	err = walkFn(root, info, err)
+	if err == filepath.SkipDir {
+		return nil
+	}
+
+	if info.IsDir() {
+		if maxDepth <= 0 {
+			return walkFn(root, info, errors.New("Filesystem bottom is too deep (directory recursion or filesystem really deep): "+root))
+		}
+		maxDepth--
+		files, err := ioutil.ReadDir(root)
+		if err == nil {
+			for _, file := range files {
+				err = simpleLocalWalkRecursive(root+string(os.PathSeparator)+file.Name(), maxDepth, walkFn)
+				if err == filepath.SkipDir {
+					return nil
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+// SimpleLocalWalk locally replaces filepath.Walk and/but goes through symlinks
+func SimpleLocalWalk(root string, walkFn func(path string, info os.FileInfo, err error) error) error {
```

Comment:
The `simpleLocalWalkRecursive()` is currently used in this module only, There's no need to wrap it and export it. Can you please remove the wrapper and use directly `simpleLocalWalkRecursive()`? 
Regarding the `maxDepth` param, can you please create a private constant in this module (named for example maxFileSystemDepth) documenting it properly writing something like: `As currently implemented on Linux, the maximum number of symbolic links that will be followed while resolving a pathname is 40`? This way you can use something more meaningful as a parameter when calling the function.
Thanks!